mesonbuild/dependencies/pkgconfig.py

```diff
@@ -73,6 +83,158 @@ def list_all(self) -> T.List[str]:
         '''Return all available pkg-config modules'''
         raise NotImplementedError
 
+    @staticmethod
+    def get_env(environment: Environment, for_machine: MachineChoice,
+                uninstalled: bool = False) -> EnvironmentVariables:
+        env = EnvironmentVariables()
+        key = OptionKey('pkg_config_path', machine=for_machine)
+        extra_paths: T.List[str] = environment.coredata.options[key].value[:]
+        if uninstalled:
+            uninstalled_path = Path(environment.get_build_dir(), 'meson-uninstalled').as_posix()
+            if uninstalled_path not in extra_paths:
+                extra_paths.append(uninstalled_path)
+        env.set('PKG_CONFIG_PATH', extra_paths)
+        sysroot = environment.properties[for_machine].get_sys_root()
+        if sysroot:
+            env.set('PKG_CONFIG_SYSROOT_DIR', [sysroot])
+        pkg_config_libdir_prop = environment.properties[for_machine].get_pkg_config_libdir()
+        if pkg_config_libdir_prop:
+            env.set('PKG_CONFIG_LIBDIR', pkg_config_libdir_prop)
+        return env
+
+
+class PyPkgConf(PkgConfigInterface):
+    '''libpkgconf implementation'''
+
+    def __init__(self, env: Environment, for_machine: MachineChoice, silent: bool):
+        self.env = env
+        self.for_machine = for_machine
+        self.pypkgconf = None
+        try:
+            self._load_pypkgconf()
+        except ImportError as e:
+            mlog.debug(str(e))
+
+    def _load_pypkgconf(self) -> None:
+        import pypkgconf
+
+        mlog.log(f'Found pypkgconf: {mlog.bold(pypkgconf.pypkgconf_version())} (libpkgconf {pypkgconf.pkgconf_version()})', once=True)
+        self.pypkgconf = pypkgconf
+
+    @functools.lru_cache(maxsize=None)
+    def query(self,
+              name: T.Optional[str] = None,
+              want_flags: T.Optional[int] = None,
+              want_variable: T.Optional[str] = None,
+              define_variable: T.Optional[T.Tuple[str, str]] = None) -> T.List[str]:
+        if self.pypkgconf is None:
+            raise MesonException('pypkgconf is not configured')
+        env = PkgConfigInterface.get_env(self.env, self.for_machine, False).get_env(os.environ)
+
+        if want_flags is None:
+            want_flags = self.pypkgconf.Flags.DEFAULT
+        args: T.Optional[T.List[str]] = [name] if name else None
+        dv = dict([define_variable]) if define_variable else None
+
+        return self.pypkgconf.query(args, want_flags=want_flags, env=env, want_variable=want_variable, define_variable=dv)
+
+    def __getstate__(self) -> object:
+        d = self.__dict__.copy()
+        del d['pypkgconf']
+        return d
+
+    def __setstate__(self, d: T.Dict) -> None:
+        self.__dict__ = d
+        try:
+            self._load_pypkgconf()
+        except ImportError:
+            raise MesonException("module pypkgconf not found")
```

Comment:
Why do you need getstate/setstate?

Reply:
Because Python modules are not pickleable